src/Tooltip.js

```diff
@@ -0,0 +1,174 @@
+import React, { Component, createRef } from 'react'
+import { createPortal } from 'react-dom'
+import propTypes from '@dhis2/prop-types'
+import { resolve } from 'styled-jsx/css'
+
+import { Popper } from './Popper.js'
+import { colors, layers } from './theme.js'
+import { referencePlacementPropType } from './common-prop-types.js'
+;('') // TODO: https://github.com/jsdoc/jsdoc/issues/1718
+
+const TOOLTIP_OFFSET = 4
+
+const popperStyle = resolve`
+    z-index: ${layers.applicationTop};
+    pointer-events: none;
+`
+
+const offsetModifier = {
+    name: 'offset',
+    options: {
+        offset: [0, TOOLTIP_OFFSET],
+    },
+}
+
+const OPEN_DELAY = 200
+const CLOSE_DELAY = 400
+
+/**
+ * @module
+ * @param {Tooltip.PropTypes} props
+ * @returns {React.Component}
+ *
+ * @example import { Tooltip } from '@dhis2/ui-core'
+ *
+ * @see Specification: {@link https://github.com/dhis2/design-system/blob/master/molecules/tooltip.md|Design system}
+ * @see Live demo: {@link /demo/?path=/story/tooltips--default-fluid|Storybook}
+ */
+class Tooltip extends Component {
+    state = { open: false }
+    ref = createRef()
+    openTimeout = null
+    closeTimeout = null
+
+    componentWillUnmount() {
+        clearTimeout(this.openTimeout)
+        clearTimeout(this.closeTimeout)
+    }
+
+    onMouseOver = () => {
+        clearTimeout(this.closeTimeout)
+
+        this.openTimeout = setTimeout(() => {
+            this.setState({ open: true })
+        }, OPEN_DELAY)
+    }
+
+    onMouseOut = () => {
+        clearTimeout(this.openTimeout)
+
+        this.closeTimeout = setTimeout(() => {
+            this.setState({ open: false })
+        }, CLOSE_DELAY)
+    }
```

Comment:
@cooper-joe only talked about a delay when entering the element (open delay), not when leaving the element (which would be a "close delay"). I also don't see a reason for delaying the hiding process.

Reply:
This should be in the updated design-specs now, and @cooper-joe already commented on this: https://github.com/dhis2/ui-core/pull/835#issuecomment-599560591. Would you mind resolving this comment?

Reply:
We are using an open delay and a close delay, so this is correct.

Close delay is used to stop tooltip flickering with unintentional mouse movements. A user moving their mouse outside of the tooltip trigger will rarely notice the delay. The delay may prevent a user with motor issues from seeing a flashing tooltip if their mouse cursor is moving quickly in/out of the trigger boundary.